transport-classes-io_uring/src/main/java/io/netty/incubator/channel/uring/IOUringEventLoop.java

```diff
@@ -154,8 +154,13 @@ protected void run() {
         final IOUringCompletionQueue completionQueue = ringBuffer.ioUringCompletionQueue();
         final IOUringSubmissionQueue submissionQueue = ringBuffer.ioUringSubmissionQueue();
 
-        // Lets add the eventfd related events before starting to do any real work.
+        // Let's add the eventfd related events before starting to do any real work.
         addEventFdRead(submissionQueue);
+        // We also need to submit this work because for short-lived event loops its possible
+        // to never enter a submitAndWait() call before shutting dwn.
+        if (submissionQueue.submit() != 1) {
+            throw new AssertionError("Failed to submit EventFdRead");
```

Comment:
Sorry to nit pick, but I'd include the return code in the error message, just on the off chance that it's something wild.